packages/command-palette-module/src/types.ts

```diff
@@ -7,19 +7,20 @@ export const SearchCollection = z.object({
 	descriptionField: z.string().optional().nullable(),
 	thumbnailField: z.string().optional().nullable(),
 	fields: z.array(z.string().min(1, 'field name cannot be empty')),
-	filter: z.object({}).optional().nullable() as z.ZodType<Filter | undefined>,
+	filter: z.object({}).optional().nullable() as z.ZodNullable<z.ZodOptional<z.ZodType<Filter>>>,
```

Comment:
This is not directly related to the upgrade, but it was causing a type error in`system-colelctions.ts`, requiring it explicitly set the `filter` field to `undefined`.